specification/powerbidedicated/Powerbidedicated.Management/models.tsp

```diff
@@ -0,0 +1,692 @@
+  state?: State;
+
+  /**
+   * The current deployment state of PowerBI Dedicated resource. The provisioningState is to indicate states for resource provisioning.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: CapacityProvisioningState;
+}
+
+/**
+ * An object that represents a set of mutable Dedicated capacity resource properties.
+ */
+model DedicatedCapacityMutableProperties {
+  /**
+   * A collection of Dedicated capacity administrators
+   */
+  administration?: DedicatedCapacityAdministrators;
+
+  /**
+   * Specifies the generation of the Power BI Embedded capacity. If no value is specified, the default value 'Gen2' is used. [Learn More](https://docs.microsoft.com/power-bi/developer/embedded/power-bi-embedded-generation-2)
+   */
+  mode?: Mode;
+
+  /**
+   * Tenant ID for the capacity. Used for creating Pro Plus capacity.
+   */
+  @visibility(Lifecycle.Read)
+  tenantId?: string;
+
+  /**
+   * Capacity name
+   */
+  @visibility(Lifecycle.Read)
+  friendlyName?: string;
+}
+
+/**
+ * An array of administrator user identities
+ */
+model DedicatedCapacityAdministrators {
+  /**
+   * An array of administrator user identities.
+   */
+  members?: string[];
+}
+
+/**
+ * Represents an instance of an PowerBI Dedicated resource.
+ */
+model Resource {
+  /**
+   * An identifier that represents the PowerBI Dedicated resource.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+
+  // /**
+  //  * The name of the PowerBI Dedicated resource.
+  //  */
+  // @visibility(Lifecycle.Read)
+  // name?: string;
+
+  /**
+   * The type of the PowerBI Dedicated resource.
+   */
+  @visibility(Lifecycle.Read)
+  type?: string;
+
+  /**
+   * Location of the PowerBI Dedicated resource.
+   */
+  @visibility(Lifecycle.Read, Lifecycle.Create)
+  location: string;
+
+  /**
+   * Key-value pairs of additional resource provisioning properties.
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  tags?: Record<string>;
+
+  /**
+   * Metadata pertaining to creation and last modification of the resource.
+   */
+  systemData?: SystemData;
+}
+
+/**
+ * Metadata pertaining to creation and last modification of the resource.
+ */
+model SystemData {
+  /**
+   * An identifier for the identity that created the resource.
+   */
+  createdBy?: string;
+
+  /**
+   * The type of identity that created the resource.
+   */
+  createdByType?: IdentityType;
+
+  /**
+   * The timestamp of resource creation (UTC).
+   */
+  createdAt?: utcDateTime;
+
+  /**
+   * An identifier for the identity that last modified the resource.
+   */
+  lastModifiedBy?: string;
+
+  /**
+   * The type of identity that last modified the resource.
+   */
+  lastModifiedByType?: IdentityType;
+
+  /**
+   * The timestamp of resource last modification (UTC).
+   */
+  lastModifiedAt?: utcDateTime;
+}
+/**
+ * The error object
+ */
+@error
+model ErrorResponse {
```

Comment:
>ErrorResponse [](http://example.com/codeflow?start=6&length=13)

why are we defining 'ErrorResponse' type here, instead of using standard Azure resource model?